internal/dataplane/translator/subtranslator/ingress.go

```diff
@@ -179,18 +242,68 @@ type ingressTranslationMeta struct {
 	ingressUID       string
 	ingressHost      string
 	ingressTags      []*string
-	serviceName      string
-	servicePort      kongstate.PortDef
+	backend          ingressTranslationMetaBackend
 	paths            []netv1.HTTPIngressPath
 	addRegexPrefixFn addRegexPrefixFn
 }
 
+type ingressPathBackendType string
+
+const (
+	ingressPathBackendTypeKongServiceFacade ingressPathBackendType = "KongServiceFacade"
+)
+
+type ingressTranslationMetaBackend struct {
+	// backendType is the type of backend. If left empty, it's assumed to be a Kubernetes Service.
+	backendType ingressPathBackendType
+
+	// name is the Kubernetes object name of the backend.
+	name string
+
+	// port is the port of the backend.
+	port kongstate.PortDef
+}
+
+func (b ingressTranslationMetaBackend) intoCacheKey(ingress k8stypes.NamespacedName, host string) string {
+	if b.backendType == ingressPathBackendTypeKongServiceFacade {
```

Comment:
So we are projecting all the ingresses with the same serviceFacade into the same Kong service? could w eadd comments to explain this?

Reply:
For every KongServiceFacade there will be one Kong Service created, that's right (even when used in multiple Ingresses, it's similar to how we handle Kubernetes Services - we create one Kong Service per  `<service-namespace>.<service-name>.<service-port>` combination, but in the case of KongServiceFacade we may omit port as only one can be defined in KongServiceFacade so it's already unique without that). 

Although, the part of code you've commented on is responsible for building a Kong Route name and it was wrong for KongServiceFacade, thanks for bringing this up. I fixed it so that separate Routes are created for every `<ingress-namespace>.<ingress-name>.<host>.<service-facade-name>` combination and left comments describing it for Service and KongServiceFacade.

PTAL d70dcef